Sources/Sentry/SentryStacktraceBuilder.m

```diff
@@ -56,6 +59,41 @@ - (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stack
     return stacktrace;
 }
 
+- (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stackCursor
+{
+    return [self retrieveStackTraceFromCursor:&stackCursor
+                        withFrameSymbolicator:^SentryFrame *(SentryCrashStackCursor *cursor) {
+                            if (stackCursor.symbolicate(cursor)) {
+                                return [self.crashStackEntryMapper mapStackEntryWithCursor:*cursor];
+                            }
+                            return nil;
+                        }];
+}
+
+- (SentryStacktrace *)retrieveStacktraceFromCursorNatively:(SentryCrashStackCursor)stackCursor
+{
+    /**
+     * Different from `retrieveStacktraceFromCursor`, this method uses `dladdr` to retrieve
+     * information from the stack cursor, which is much faster and thread safe but is not async
+     * safe, that's why this method cannot be used during crashes.
+     */
+
+    return [self retrieveStackTraceFromCursor:&stackCursor
+                        withFrameSymbolicator:^SentryFrame *(SentryCrashStackCursor *cursor) {
+                            Dl_info info = { 0 };
+                            if (dladdr((const void *)cursor->stackEntry.address, &info)) {
```

Comment:
`h`: Why don't we use `sentrycrashsymbolicator_symbolicate` here? The code looks almost the same and it has support for async markers.

Reply:
Because is super slow.

Reply:
Yes, but only because it calls into `sentrycrashdl_dladdr`. What about passing in a flag to `sentrycrashsymbolicator_symbolicate` to control whether to call `sentrycrashdl_dladdr` or `dladdr` or refactoring it to use the same code?

Reply:
Got it. I'll take a look whether this is possible,

Reply:
If you dont mind I will do this refactoring in a follow up PR. We need this feature to be available with v8 if possible.

Reply:
When you open an issue so we don't forget about it. It's acceptable.

Reply:
[Issue](https://github.com/getsentry/sentry-cocoa/issues/2602)